File: C/Lab7/sort.c
#include <stdio.h>

int bubbleSort(int *array, int size);

int bubbleSort(int *array, int size){
    if (size<1) {
        return -1;
    }
    if (array == NULL) {
        return -1;
    }
    int n = size;
    int swapping = 0;
    while (swapping == 0){
        int swapped = 0;
        int i;
        for (i=1;i<n;i+=1){
            if (array[i-1]>array[i]) {
                int temp = array[i-1];
                array[i-1] = array[i];
                array[i] = temp;
                swapped = 1;
            }
        }
        n = n-1;
        if (swapped == 0){
            swapping = 1;
        }
    }
    return 0;
}

File: C/Lab7/sort.py
def bubbleSort(array):
    n = len(array)
    swapping = True
    try:
        while swapping == True:
            swapped = False
            for i in range(1,n,1):
                if array[i-1] > array[i]:
                    temp = array[i-1]
                    array[i-1] = array[i]
                    array[i] = temp
                    swapped = True
            n = n-1
            if not swapped:
                swapping = False
        return [True, array]
    except:
        return False
